### ISSUE TITLE
kic: fix var race condition

### ISSUE THREAD

**Issue by medyagh** [kubernetes/minikube#8657]

**Comment by afbjorklund** [comment]
The race condition is because we have `/var` both in the kicbase image (from ubuntu), **and** `/var` mounted as a volume.

And if the volume wins the race, then we won't copy those files from the image to the volume - and miss some important ones...

**Comment by afbjorklund** [comment]
This is related to #8151 and #8458 as well, since we don't mount the directories needed for the storage provisioner.

But if we add a `minikube-automount` to the KIC, then it can take care of adding all of these directories to the volume.

**Comment by medyagh** [comment]
dupe of https://github.com/kubernetes/minikube/issues/8458

**Comment by afbjorklund** [comment]
The #8458 and #8151 are about making sure that the storage provisioner mounts the correct directories.
It is similar to the same issue for the "none" driver, where the user is supposed to mount them: #7511

However, _this_ particular issue is about the race condition that happens between "start" and "preload".
We have one container that uses a named volume (copying contents), and the creation of that volume.


So we are going to fix the storage mounts in that other issue, and this is about fixing the start up order.
Initially I was going to change the `/var` mount point, but since that will break with KIND we will delay that.

Instead we are going to make sure that the volume is created by "docker run", and not create an empty one.
This way we make sure that the image contents are present, such as /var/lib/dpkg/alternative which is needed.

It's a bit unfortunate that we opened a new "blank" story for the new milestone, but the issue is: #8179 #8100

We saw this already with the podman driver, but there the workaround was to disable the parallell preload: #8056

So the error we are trying to fix here, is the bootup error:

`update-alternatives: error: no alternatives for iptables`

Either we do a simple "docker run true" before the tar run.

Or we have some more pretty way to make sure it happens...
